omod/src/main/java/org/openmrs/web/controller/maintenance/SearchIndexController.java

```diff
@@ -47,13 +50,39 @@ public String showPage() {
 		boolean success = true;
 		Map<String, Object> results = new HashMap<String, Object>();
 		log.debug("rebuilding search index");
-		try {
-			Context.updateSearchIndex();
-		} catch (RuntimeException e) {
+		if (!Context.getUserContext().isAuthenticated()) {
 			success = false;
```

Comment:
So when a non authenticated user tries to build the index, does it just fail, or does it tell them that they have to first login, or does it redirect them to the login page?